### ISSUE TITLE
cmake and make problems on openSUSE Tumbleweed

### ISSUE THREAD

**Issue by cjlee13** [rilian-la-te/vala-panel-appmenu#210]
I switch from Ubuntu to openSUSE Tumbleweed recently. Since I cannot found newer version of this package on repos, I try to built it myself. However, when trying to compile it, I encountered `fatal error: wayland-client.h: No such file or directory`. But I have it at /usr/include/wayland/wayland-client.h. 
Also, cmake can't find bamfdaemon until I manually assign the path to in by `-DBAMF_DAEMON_EXECUTABLE=/usr/lib/bamf `

I really appreciate your work. Thank you!

This is the output of cmake and make:
```
cjlee@linux-5vcj:~/vala-panel-appmenu/build> cmake -DENABLE_XFCE=ON -DENABLE_BUDGIE=OFF -DENABLE_VALAPANEL=OFF -DENABLE_MATE=OFF -DENABLE_JAYATANA=ON -DENABLE_APPMENU_GTK_MODULE=ON -DMAKE_BOLD_APPNAME=ON -DBAMF_DAEMON_EXECUTABLE=/usr/lib/bamf ..
-- Looking for RPMTools... - found rpmuild is /usr/bin/rpmbuild
-- RPMTools:: Using RPM_ROOTDIR=/home/cjlee/vala-panel-appmenu/build/RPM
-- Configuring spec file </home/cjlee/vala-panel-appmenu/rpm/vala-panel-appmenu.spec.in> --> </home/cjlee/vala-panel-appmenu/build/RPM/SPECS/vala-panel-appmenu.spec>
-- RPMTools:: Using RPM_ROOTDIR=/home/cjlee/vala-panel-appmenu/build/RPM
-- Configuring spec file </home/cjlee/vala-panel-appmenu/rpm/appmenu-gtk-module.spec.in> --> </home/cjlee/vala-panel-appmenu/build/RPM/SPECS/appmenu-gtk-module.spec>
-- RPMTools:: Using RPM_ROOTDIR=/home/cjlee/vala-panel-appmenu/build/RPM
-- Configuring spec file </home/cjlee/vala-panel-appmenu/rpm/java-appmenu.spec.in> --> </home/cjlee/vala-panel-appmenu/build/RPM/SPECS/java-appmenu.spec>
-- Found GTK: /usr/lib64/libgdk-3.so;/usr/lib64/libgtk-3.so (found suitable version "3.22.26", minimum required is "3.22") found components:  GDK GTK X11 WAYLAND 
Using libwnck, will be X-dependent
-- Checking for module 'libmatepanelapplet-4.0'
--   No package 'libmatepanelapplet-4.0' found
-- Could NOT find PEAS: Found unsuitable version "", but required is at least "1.14" (found )
-- Checking for module 'vala-panel'
--   No package 'vala-panel' found
-- Checking for module 'budgie-1.0'
--   No package 'budgie-1.0' found
-- Found GTK: /usr/lib64/libgdk-x11-2.0.so;/usr/lib64/libgtk-x11-2.0.so (found suitable version "2.24.32", minimum required is "2.24") found components:  GDK GTK X11 
-- GSettings schemas will be installed into /usr/local/share/glib-2.0/schemas/
-- Could NOT find Java (missing: Java_JAR_EXECUTABLE Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE) (found version "1.8.0_151")
-- Could NOT find JNI (missing: JAVA_INCLUDE_PATH JAVA_INCLUDE_PATH2 JAVA_AWT_INCLUDE_PATH) 
-- Checking for module 'dbusmenu-glib-0.4>=16.04.0'
--   No package 'dbusmenu-glib-0.4' found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/cjlee/vala-panel-appmenu/build
cjlee@linux-5vcj:~/vala-panel-appmenu/build> make
[  9%] Built target appmenu-gtk2-parser
[ 18%] Built target appmenu-gtk3-parser
[ 29%] Built target appmenu-gtk2-module
[ 31%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/datastructs.c.o
In file included from /usr/include/gtk-3.0/gdk/gdkwayland.h:32:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/src/platform.h:33,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/src/datastructs.c:27:
/usr/include/gtk-3.0/gdk/wayland/gdkwaylanddevice.h:27:10: fatal error: wayland-client.h: No such file or directory
 #include <wayland-client.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/build.make:87: subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/datastructs.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:576: subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
cjlee@linux-5vcj:~/vala-panel-appmenu/build> ll -a /usr/include/wayland/
total 388
drwxr-xr-x 1 root root    404 Feb 21 06:27 .
drwxr-xr-x 1 root root   3866 Feb 21 16:43 ..
-rw-r--r-- 1 root root   8615 Oct 23 23:35 wayland-client-core.h
-rw-r--r-- 1 root root   1573 Oct 23 23:35 wayland-client.h
-rw-r--r-- 1 root root 177539 Oct 23 23:35 wayland-client-protocol.h
-rw-r--r-- 1 root root   2243 Oct 23 23:35 wayland-cursor.h
-rw-r--r-- 1 root root   1788 Oct 23 23:35 wayland-egl-core.h
-rw-r--r-- 1 root root   1313 Oct 23 23:35 wayland-egl.h
-rw-r--r-- 1 root root  16231 Oct 23 23:35 wayland-server-core.h
-rw-r--r-- 1 root root   2929 Oct 23 23:35 wayland-server.h
-rw-r--r-- 1 root root 138870 Oct 23 23:35 wayland-server-protocol.h
-rw-r--r-- 1 root root  23111 Oct 23 23:35 wayland-util.h
-rw-r--r-- 1 root root   1354 Oct 23 23:35 wayland-version.h
cjlee@linux-5vcj:~/vala-panel-appmenu/build> 
```

**Comment by rilian-la-te** [comment]
Will look for it.

**Comment by rilian-la-te** [comment]
Try now)

**Comment by rilian-la-te** [comment]
If working, please close)

**Comment by cjlee13** [comment]
Sorry, I got the same error message.

I'm using 64-bit Tumbleweed. Is this relevant?
But I had already tried installing wayland-devel-32-bit. It didn't help.

**Comment by rilian-la-te** [comment]
It should work, AFAIK. Can you recreate build dir and send me CMakeCache.txt, CMake log and new build log?

**Comment by cjlee13** [comment]
[CMakeCache.txt](https://github.com/rilian-la-te/vala-panel-appmenu/files/1755716/CMakeCache.txt)

Strange. I got more messages after packages update today.
```
cjlee@linux-5vcj:~/vala-panel-appmenu/build> cmake -DENABLE_XFCE=ON -DENABLE_BUDGIE=OFF -DENABLE_VALAPANEL=OFF -DENABLE_MATE=OFF -DENABLE_JAYATANA=ON -DENABLE_APPMENU_GTK_MODULE=ON -DBAMF_DAEMON_EXECUTABLE=/usr/lib/bamf ..
-- The C compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for RPMTools... - found rpmuild is /usr/bin/rpmbuild
-- RPMTools:: Using RPM_ROOTDIR=/home/cjlee/vala-panel-appmenu/build/RPM
-- Configuring spec file </home/cjlee/vala-panel-appmenu/rpm/vala-panel-appmenu.spec.in> --> </home/cjlee/vala-panel-appmenu/build/RPM/SPECS/vala-panel-appmenu.spec>
-- RPMTools:: Using RPM_ROOTDIR=/home/cjlee/vala-panel-appmenu/build/RPM
-- Configuring spec file </home/cjlee/vala-panel-appmenu/rpm/appmenu-gtk-module.spec.in> --> </home/cjlee/vala-panel-appmenu/build/RPM/SPECS/appmenu-gtk-module.spec>
-- RPMTools:: Using RPM_ROOTDIR=/home/cjlee/vala-panel-appmenu/build/RPM
-- Configuring spec file </home/cjlee/vala-panel-appmenu/rpm/java-appmenu.spec.in> --> </home/cjlee/vala-panel-appmenu/build/RPM/SPECS/java-appmenu.spec>
-- Found Vala: /usr/bin/valac (found suitable version "0.38.8", minimum required is "0.34") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Found GLIB2: /usr/lib64/libglib-2.0.so;/usr/lib64/libgobject-2.0.so;/usr/lib64/libgio-2.0.so;/usr/lib64/libgio-2.0.so;/usr/lib64/libgmodule-2.0.so;/usr/lib64/libgthread-2.0.so (found suitable version "2.54.3", minimum required is "2.50") found components:  GOBJECT GIO GIO_UNIX GMODULE GTHREAD CODEGEN MKENUMS GENMARSHAL COMPILE_SCHEMAS COMPILE_RESOURCES 
-- Found GTK: /usr/lib64/libgdk-3.so;/usr/lib64/libgtk-3.so (found suitable version "3.22.26", minimum required is "3.22") found components:  GDK GTK X11 WAYLAND 
-- Found BAMF: TRUE (found suitable version "0.5.3", minimum required is "0.5.0") found components:  LIB DAEMON 
-- Found WNCK: /usr/lib64/libwnck-3.so (found suitable version "3.24.1", minimum required is "3.4.8") 
Using libwnck, will be X-dependent
-- Checking for modules 'libxfce4panel-2.0;libxfconf-0'
--   Found libxfce4panel-2.0, version 4.12.2
--   Found libxfconf-0, version 4.12.1
-- Checking for module 'libmatepanelapplet-4.0'
--   No package 'libmatepanelapplet-4.0' found
-- Could NOT find PEAS: Found unsuitable version "", but required is at least "1.14" (found )
-- Checking for module 'vala-panel'
--   No package 'vala-panel' found
-- Checking for module 'budgie-1.0'
--   No package 'budgie-1.0' found
-- Found GTK: /usr/lib64/libgdk-x11-2.0.so;/usr/lib64/libgtk-x11-2.0.so (found suitable version "2.24.32", minimum required is "2.24") found components:  GDK GTK X11 
-- GSettings schemas will be installed into /usr/local/share/glib-2.0/schemas/
-- Checking for module 'systemd'
--   Found systemd, version 234
-- Could NOT find Java (missing: Java_JAR_EXECUTABLE Java_JAVAC_EXECUTABLE Java_JAVAH_EXECUTABLE Java_JAVADOC_EXECUTABLE) (found version "1.8.0_151")
-- Could NOT find JNI (missing: JAVA_INCLUDE_PATH JAVA_INCLUDE_PATH2 JAVA_AWT_INCLUDE_PATH) 
-- Checking for module 'xkbcommon'
--   Found xkbcommon, version 0.8.0
-- Checking for module 'dbusmenu-glib-0.4>=16.04.0'
--   No package 'dbusmenu-glib-0.4' found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/cjlee/vala-panel-appmenu/build
cjlee@linux-5vcj:~/vala-panel-appmenu/build> make
Scanning dependencies of target appmenu-gtk2-parser
[  1%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk2-parser.dir/unity-gtk-action-group.c.o
[  3%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk2-parser.dir/unity-gtk-action.c.o
[  4%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk2-parser.dir/unity-gtk-menu-shell.c.o
[  6%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk2-parser.dir/unity-gtk-menu-section.c.o
[  7%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk2-parser.dir/unity-gtk-menu-item.c.o
[  9%] Linking C shared library libappmenu-gtk2-parser.so
[  9%] Built target appmenu-gtk2-parser
Scanning dependencies of target appmenu-gtk3-parser
[ 10%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk3-parser.dir/unity-gtk-action-group.c.o
[ 12%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk3-parser.dir/unity-gtk-action.c.o
[ 14%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk3-parser.dir/unity-gtk-menu-shell.c.o
[ 15%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk3-parser.dir/unity-gtk-menu-section.c.o
[ 17%] Building C object subprojects/appmenu-gtk-module/lib/CMakeFiles/appmenu-gtk3-parser.dir/unity-gtk-menu-item.c.o
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c: In function ‘gtk_image_get_icon’:
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:185:4: warning: ‘gtk_image_get_stock’ is deprecated: Use 'gtk_image_get_icon_name' instead [-Wdeprecated-declarations]
    gtk_image_get_stock(image, &stock, &size);
    ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtkentry.h:42:0,
                 from /usr/include/gtk-3.0/gtk/gtktreeview.h:29,
                 from /usr/include/gtk-3.0/gtk/gtkcombobox.h:27,
                 from /usr/include/gtk-3.0/gtk/gtkappchooserbutton.h:29,
                 from /usr/include/gtk-3.0/gtk/gtk.h:42,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/gtkimage.h:181:12: note: declared here
 void       gtk_image_get_stock    (GtkImage         *image,
            ^~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:189:5: warning: ‘gtk_style_context_lookup_icon_set’ is deprecated: Use 'gtk_icon_theme_lookup_icon' instead [-Wdeprecated-declarations]
     GtkIconSet *set = gtk_style_context_lookup_icon_set(context, stock);
     ^~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtkicontheme.h:27:0,
                 from /usr/include/gtk-3.0/gtk/gtk.h:123,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/gtkstylecontext.h:1131:14: note: declared here
 GtkIconSet * gtk_style_context_lookup_icon_set (GtkStyleContext *context,
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:194:10: warning: ‘gtk_icon_set_render_icon_pixbuf’ is deprecated [-Wdeprecated-declarations]
          gtk_icon_set_render_icon_pixbuf(set, context, size);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtkicontheme.h:27:0,
                 from /usr/include/gtk-3.0/gtk/gtk.h:123,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/gtkstylecontext.h:1134:14: note: declared here
 GdkPixbuf  * gtk_icon_set_render_icon_pixbuf   (GtkIconSet      *icon_set,
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:240:4: warning: ‘gtk_image_get_icon_set’ is deprecated: Use 'gtk_image_get_icon_name' instead [-Wdeprecated-declarations]
    gtk_image_get_icon_set(image, &set, &size);
    ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtkentry.h:42:0,
                 from /usr/include/gtk-3.0/gtk/gtktreeview.h:29,
                 from /usr/include/gtk-3.0/gtk/gtkcombobox.h:27,
                 from /usr/include/gtk-3.0/gtk/gtkappchooserbutton.h:29,
                 from /usr/include/gtk-3.0/gtk/gtk.h:42,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/gtkimage.h:185:12: note: declared here
 void       gtk_image_get_icon_set (GtkImage         *image,
            ^~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:245:9: warning: ‘gtk_icon_set_render_icon_pixbuf’ is deprecated [-Wdeprecated-declarations]
         gtk_icon_set_render_icon_pixbuf(set, context, size);
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtkicontheme.h:27:0,
                 from /usr/include/gtk-3.0/gtk/gtk.h:123,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/gtkstylecontext.h:1134:14: note: declared here
 GdkPixbuf  * gtk_icon_set_render_icon_pixbuf   (GtkIconSet      *icon_set,
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c: In function ‘unity_gtk_menu_item_get_label’:
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:745:4: warning: ‘gtk_image_menu_item_get_type’ is deprecated: Use 'gtk_menu_item_get_type' instead [-Wdeprecated-declarations]
    if (GTK_IS_IMAGE_MENU_ITEM(item->menu_item))
    ^~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:265:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:76:10: note: declared here
 GType    gtk_image_menu_item_get_type          (void) G_GNUC_CONST;
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:748:9: warning: ‘gtk_image_menu_item_get_type’ is deprecated: Use 'gtk_menu_item_get_type' instead [-Wdeprecated-declarations]
         GTK_IMAGE_MENU_ITEM(item->menu_item);
         ^~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:265:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:76:10: note: declared here
 GType    gtk_image_menu_item_get_type          (void) G_GNUC_CONST;
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:750:5: warning: ‘gtk_image_menu_item_get_use_stock’ is deprecated [-Wdeprecated-declarations]
     if (gtk_image_menu_item_get_use_stock(image_menu_item))
     ^~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:265:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:100:12: note: declared here
 gboolean   gtk_image_menu_item_get_use_stock     (GtkImageMenuItem *image_menu_item);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:754:6: warning: ‘gtk_stock_lookup’ is deprecated [-Wdeprecated-declarations]
      if (gtk_stock_lookup(label_label, &stock_item))
      ^~
In file included from /usr/include/gtk-3.0/gtk/deprecated/gtkactiongroup.h:37:0,
                 from /usr/include/gtk-3.0/gtk/gtk.h:250,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/deprecated/gtkstock.h:82:10: note: declared here
 gboolean gtk_stock_lookup     (const gchar         *stock_id,
          ^~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c: In function ‘unity_gtk_menu_item_get_icon’:
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:788:2: warning: ‘gtk_image_menu_item_get_type’ is deprecated: Use 'gtk_menu_item_get_type' instead [-Wdeprecated-declarations]
  if (item->menu_item != NULL && !GTK_IS_IMAGE_MENU_ITEM(item->menu_item))
  ^~
In file included from /usr/include/gtk-3.0/gtk/gtk.h:265:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/deprecated/gtkimagemenuitem.h:76:10: note: declared here
 GType    gtk_image_menu_item_get_type          (void) G_GNUC_CONST;
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c: In function ‘unity_gtk_menu_item_is_visible’:
/home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:804:9: warning: ‘gtk_tearoff_menu_item_get_type’ is deprecated [-Wdeprecated-declarations]
         !GTK_IS_TEAROFF_MENU_ITEM(item->menu_item);
         ^
In file included from /usr/include/gtk-3.0/gtk/gtk.h:277:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-menu-shell.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/appmenu-gtk-action-group.h:44,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-action-group-private.h:23,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/lib/unity-gtk-menu-item.c:20:
/usr/include/gtk-3.0/gtk/deprecated/gtktearoffmenuitem.h:75:10: note: declared here
 GType    gtk_tearoff_menu_item_get_type     (void) G_GNUC_CONST;
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 18%] Linking C shared library libappmenu-gtk3-parser.so
[ 18%] Built target appmenu-gtk3-parser
Scanning dependencies of target appmenu-gtk2-module
[ 20%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk2-module.dir/appmenu-gtk-module.c.o
[ 21%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk2-module.dir/datastructs.c.o
[ 23%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk2-module.dir/hijack.c.o
[ 25%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk2-module.dir/support.c.o
[ 26%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk2-module.dir/blacklist.c.o
[ 28%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk2-module.dir/platform.c.o
[ 29%] Linking C shared module gtk-2.0/libappmenu-gtk-module.so
[ 29%] Built target appmenu-gtk2-module
Scanning dependencies of target appmenu-gtk3-module
[ 31%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/appmenu-gtk-module.c.o
[ 32%] Building C object subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/datastructs.c.o
In file included from /usr/include/gtk-3.0/gdk/gdkwayland.h:32:0,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/src/platform.h:33,
                 from /home/cjlee/vala-panel-appmenu/subprojects/appmenu-gtk-module/src/datastructs.c:27:
/usr/include/gtk-3.0/gdk/wayland/gdkwaylanddevice.h:27:10: fatal error: wayland-client.h: No such file or directory
 #include <wayland-client.h>
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/build.make:87: subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/datastructs.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:576: subprojects/appmenu-gtk-module/src/CMakeFiles/appmenu-gtk3-module.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
cjlee@linux-5vcj:~/vala-panel-appmenu/build>
```

**Comment by rilian-la-te** [comment]
Seems fixed.
